### PR TITLE
feat: Move all the configs to one single file

### PR DESCRIPTION
**Pull request by Natoandro**
- Move all the configs to one single file
- Remove some specific configs from the global config variable and make them accessible on the `Typegate` instance, to improve test configurability.

#### Migration notes

- `SYNC_REDIS_PASSWORD` has been removed, can only be set on the `SYNC_REDIS_URL`.

- [x] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [x] End-user documentation is updated to reflect the change

### CODE REVIEW

File: typegate/src/sync/config.ts [hunk 1]
```diff
@@ -1,135 +1,135 @@
...
+// export async function syncConfigFromEnv(
+//   sources: ConfigSource[],
+// ): Promise<SyncConfig | null> {
+//   const rawObjects = sources.map((source) => {
+//     switch (source) {
+//       case "vars":
+//         return mapKeys(
+//           Deno.env.toObject(),
+//           (k: string) => k.toLowerCase(),
+//         );
```
Comment (zifeo) [inline typegate/src/sync/config.ts:127]: comments on purpose?

Comment (Natoandro) [reply typegate/src/sync/config.ts:127]: Will be removed...

File: whiz.yaml [hunk 2]
```diff
@@ -19,7 +19,7 @@ gate1: &tp
     REDIS_URL: "redis://:password@localhost:6379/0"
     TG_SECRET: "a4lNi0PbEItlFZbus1oeH/+wyIxi9uH6TpL8AIqIaMBNvp7SESmuUBbfUwC0prxhGhZqHw8vMDYZAGMhSZ4fLw=="
     TG_ADMIN_PASSWORD: "password"
-    LD_LIBRARY_PATH: /data/home/metatype/.ghjk/envs/default/shims/lib
```
Comment (zifeo) [inline whiz.yaml:22]: is that wanted?

Comment (Natoandro) [reply whiz.yaml:22]: The path absolute. It was specific to his setup when he added it there...

Comment (zifeo) [reply whiz.yaml:22]: Yeah, should it be `./.ghjk/envs/default/shims/lib` instead?

Comment (Yohe-Am) [reply whiz.yaml:22]: Yeah, this is no longer necessary since #754 .

**Review by Yohe-Am — APPROVED** [review]
Good diff.

**Review by Yohe-Am — APPROVED** [review]

**Pull request merged**